src/services/offer.js

```diff
@@ -227,15 +227,15 @@ class OfferService {
         return constraints.length ? { "$and": constraints } : {};
     }
 
-    async getOfferById(offerId, user, showAdminReason = false) {
+    async getOfferById(offerId, req, showAdminReason = false) {
```

Comment:
The function should not receive the entire request object, but only the boolean `req.hasAdminPrivileges`. You can call the arg `hasAdminPrivileges` to make it easy to understand.

This is based on the premise that service functions shouldn't need to know about requests or routing or whatever, we could easily call them aside from that (which makes it easier for testing for example)